### ISSUE TITLE
Left Half not connecting 

### ISSUE THREAD

**Issue by svenknoke** [UltimateHackingKeyboard/agent#1629]
# Tried to update fw to 8.10.12. Aborted due to left half not connected issues

## symptoms
- left half is dead (not shown in agent, no keystrokes to computer, leds off)
- error messages during flashing (see log below)

## Versions
UHK V1 
agent updated to 1.5.16  from 1.5.12 (i _think not sure as i didn't write old version down)
previous fw 8.1.8 (that what i definitely rember :-)  )

##I tried to recover by the following methods

- reconnecting left half (cable disconnect / reconnect) during flash process according to agent messages
- reconnecting left half (separate and rejoin keyboard halves) during flash process according to agent messages
- flashing downloaded fw v 8.1.0 according to issue 74
- use an older agent (1.5.15) 
- use windows version of agent
- use linux version of agent

all tries without success 

## log
Start flashing firmware
[DeviceService] Operating system: Linux 5.11.16-arch1-1 x64
[DeviceService] Agent version: 1.5.16
[DeviceService] New firmware version: 8.10.12
[DeviceService] stop poll UHK Device
[DeviceService] stopped poll UHK Device
[DeviceService] UHK Device firmware upgrade starts: {"id":1,"name":"UHK 60 v1","vendorId":"0x1D50","keyboardPid":"0x6122","bootloaderPid":"0x6120","buspalPid":"0x6121"}
[DeviceService] Device right firmware version: 8.10.12
Skip right firmware upgrade.
[DeviceService] Left module firmware version:  null
[UhkOperations] Start flashing "Left keyboard half" module firmware
[UhkHidDevice] Start reenumeration, mode: NormalKeyboard, timeout: 5000ms
[UhkHidDevice] Reenumerating devices
[UhkHidDevice] Device communication closing.
[UhkHidDevice] Device communication closed.
[UhkHidDevice] Available devices unchanged
[UhkHidDevice] Used device: {"vendorId":"0x1D50","productId":"0x6122","path":"/dev/hidraw0","serialNumber":"","manufacturer":"Ultimate Gadget Laboratories","product":"UHK 60 v1","release":258,"interface":0,"usagePage":65280,"usage":1}
[UhkHidDevice] Device communication closing.
[UhkHidDevice] Device communication closed.
[UhkHidDevice] Available devices unchanged
[UhkHidDevice] Used device: {"vendorId":"0x1D50","productId":"0x6122","path":"/dev/hidraw0","serialNumber":"","manufacturer":"Ultimate Gadget Laboratories","product":"UHK 60 v1","release":258,"interface":0,"usagePage":65280,"usage":1}
[UhkHidDevice] Device communication closing.
[UhkHidDevice] Device communication closed.
[DeviceOperation] Cannot ping the bootloader. Please remove the "Left keyboard half" module, and keep reconnecting it until you see this message.
[UhkHidDevice] Available devices unchanged
[UhkHidDevice] Used device: {"vendorId":"0x1D50","productId":"0x6122","path":"/dev/hidraw0","serialNumber":"","manufacturer":"Ultimate Gadget Laboratories","product":"UHK 60 v1","release":258,"interface":0,"usagePage":65280,"usage":1}
[UhkHidDevice] Device communication closing.
[UhkHidDevice] Device communication closed.
[DeviceOperation] Cannot ping the bootloader. Please remove the "Left keyboard half" module, and keep reconnecting it until you see this message.
[UhkHidDevice] Available devices unchanged
[UhkHidDevice] Used device: {"vendorId":"0x1D50","productId":"0x6122","path":"/dev/hidraw0","serialNumber":"","manufacturer":"Ultimate Gadget Laboratories","product":"UHK 60 v1","release":258,"interface":0,"usagePage":65280,"usage":1}
[UhkHidDevice] Device communication closing.
[UhkHidDevice] Device communication closed.
[DeviceOperation] Cannot ping the bootloader. Please remove the "Left keyboard half" module, and keep reconnecting it until you see this message.
[UhkHidDevice] Available devices unchanged
[UhkHidDevice] Used device: {"vendorId":"0x1D50","productId":"0x6122","path":"/dev/hidraw0","serialNumber":"","manufacturer":"Ultimate Gadget Laboratories","product":"UHK 60 v1","release":258,"interface":0,"usagePage":65280,"usage":1}
[UhkHidDevice] Device communication closing.
[UhkHidDevice] Device communication closed.
[DeviceOperation] Cannot ping the bootloader. Please remove the "Left keyboard half" module, and keep reconnecting it until you see this message.
[UhkHidDevice] Available devices unchanged
[UhkHidDevice] Used device: {"vendorId":"0x1D50","productId":"0x6122","path":"/dev/hidraw0","serialNumber":"","manufacturer":"Ultimate Gadget Laboratories","product":"UHK 60 v1","release":258,"interface":0,"usagePage":65280,"usage":1}
[UhkHidDevice] Device communication closing.
[UhkHidDevice] Device communication closed.
[DeviceOperation] Cannot ping the bootloader. Please remove the "Left keyboard half" module, and keep reconnecting it until you see this message.
[UhkHidDevice] Available devices unchanged
[UhkHidDevice] Used device: {"vendorId":"0x1D50","productId":"0x6122","path":"/dev/hidraw0","serialNumber":"","manufacturer":"Ultimate Gadget Laboratories","product":"UHK 60 v1","release":258,"interface":0,"usagePage":65280,"usage":1}
[UhkHidDevice] Device communication closing.
[UhkHidDevice] Device communication closed.
[DeviceOperation] Cannot ping the bootloader. Please remove the "Left keyboard half" module, and keep reconnecting it until you see this message.
[UhkHidDevice] Available devices unchanged
[UhkHidDevice] Used device: {"vendorId":"0x1D50","productId":"0x6122","path":"/dev/hidraw0","serialNumber":"","manufacturer":"Ultimate Gadget Laboratories","product":"UHK 60 v1","release":258,"interface":0,"usagePage":65280,"usage":1}
[UhkHidDevice] Device communication closing.
[UhkHidDevice] Device communication closed.
[DeviceOperation] Cannot ping the bootloader. Please remove the "Left keyboard half" module, and keep reconnecting it until you see this message.
[UhkHidDevice] Available devices unchanged
[UhkHidDevice] Used device: {"vendorId":"0x1D50","productId":"0x6122","path":"/dev/hidraw0","serialNumber":"","manufacturer":"Ultimate Gadget Laboratories","product":"UHK 60 v1","release":258,"interface":0,"usagePage":65280,"usage":1}
[UhkHidDevice] Device communication closing.
[UhkHidDevice] Device communication closed.
[DeviceOperation] Cannot ping the bootloader. Please remove the "Left keyboard half" module, and keep reconnecting it until you see this message.
[UhkHidDevice] Available devices unchanged
[UhkHidDevice] Used device: {"vendorId":"0x1D50","productId":"0x6122","path":"/dev/hidraw0","serialNumber":"","manufacturer":"Ultimate Gadget Laboratories","product":"UHK 60 v1","release":258,"interface":0,"usagePage":65280,"usage":1}
[UhkHidDevice] Device communication closing.
[UhkHidDevice] Device communication closed.
[DeviceOperation] Cannot ping the bootloader. Please remove the "Left keyboard half" module, and keep reconnecting it until you see this message.
[UhkHidDevice] Available devices unchanged
[UhkHidDevice] Used device: {"vendorId":"0x1D50","productId":"0x6122","path":"/dev/hidraw0","serialNumber":"","manufacturer":"Ultimate Gadget Laboratories","product":"UHK 60 v1","release":258,"interface":0,"usagePage":65280,"usage":1}
[UhkHidDevice] Device communication closing.
[UhkHidDevice] Device communication closed.
[DeviceOperation] Cannot ping the bootloader. Please remove the "Left keyboard half" module, and keep reconnecting it until you see this message.
[UhkHidDevice] Available devices unchanged
[UhkHidDevice] Used device: {"vendorId":"0x1D50","productId":"0x6122","path":"/dev/hidraw0","serialNumber":"","manufacturer":"Ultimate Gadget Laboratories","product":"UHK 60 v1","release":258,"interface":0,"usagePage":65280,"usage":1}
[UhkHidDevice] Device communication closing.
[UhkHidDevice] Device communication closed.
[DeviceOperation] Cannot ping the bootloader. Please remove the "Left keyboard half" module, and keep reconnecting it until you see this message.
[UhkHidDevice] Available devices unchanged
[UhkHidDevice] Used device: {"vendorId":"0x1D50","productId":"0x6122","path":"/dev/hidraw0","serialNumber":"","manufacturer":"Ultimate Gadget Laboratories","product":"UHK 60 v1","release":258,"interface":0,"usagePage":65280,"usage":1}
[UhkHidDevice] Device communication closing.
[UhkHidDevice] Device communication closed.
[DeviceOperation] Cannot ping the bootloader. Please remove the "Left keyboard half" module, and keep reconnecting it until you see this message.
[UhkHidDevice] Available devices unchanged
[UhkHidDevice] Used device: {"vendorId":"0x1D50","productId":"0x6122","path":"/dev/hidraw0","serialNumber":"","manufacturer":"Ultimate Gadget Laboratories","product":"UHK 60 v1","release":258,"interface":0,"usagePage":65280,"usage":1}
[UhkHidDevice] Device communication closing.
[UhkHidDevice] Device communication closed.
[DeviceOperation] Cannot ping the bootloader. Please remove the "Left keyboard half" module, and keep reconnecting it until you see this message.
[UhkHidDevice] Available devices unchanged
[UhkHidDevice] Used device: {"vendorId":"0x1D50","productId":"0x6122","path":"/dev/hidraw0","serialNumber":"","manufacturer":"Ultimate Gadget Laboratories","product":"UHK 60 v1","release":258,"interface":0,"usagePage":65280,"usage":1}
[UhkHidDevice] Device communication closing.
[UhkHidDevice] Device communication closed.
[DeviceOperation] Cannot ping the bootloader. Please remove the "Left keyboard half" module, and keep reconnecting it until you see this message.
[UhkHidDevice] Available devices unchanged
[UhkHidDevice] Used device: {"vendorId":"0x1D50","productId":"0x6122","path":"/dev/hidraw0","serialNumber":"","manufacturer":"Ultimate Gadget Laboratories","product":"UHK 60 v1","release":258,"interface":0,"usagePage":65280,"usage":1}
[UhkHidDevice] Device communication closing.
[UhkHidDevice] Device communication closed.
[DeviceOperation] Cannot ping the bootloader. Please remove the "Left keyboard half" module, and keep reconnecting it until you see this message.
[UhkHidDevice] Available devices unchanged
[UhkHidDevice] Used device: {"vendorId":"0x1D50","productId":"0x6122","path":"/dev/hidraw0","serialNumber":"","manufacturer":"Ultimate Gadget Laboratories","product":"UHK 60 v1","release":258,"interface":0,"usagePage":65280,"usage":1}
[UhkHidDevice] Device communication closing.
[UhkHidDevice] Device communication closed.
[DeviceOperation] Cannot ping the bootloader. Please remove the "Left keyboard half" module, and keep reconnecting it until you see this message.
[UhkHidDevice] Available devices unchanged
[UhkHidDevice] Used device: {"vendorId":"0x1D50","productId":"0x6122","path":"/dev/hidraw0","serialNumber":"","manufacturer":"Ultimate Gadget Laboratories","product":"UHK 60 v1","release":258,"interface":0,"usagePage":65280,"usage":1}
[UhkHidDevice] Device communication closing.
[UhkHidDevice] Device communication closed.
[DeviceOperation] Cannot ping the bootloader. Please remove the "Left keyboard half" module, and keep reconnecting it until you see this message.
[UhkHidDevice] Available devices unchanged
[UhkHidDevice] Used device: {"vendorId":"0x1D50","productId":"0x6122","path":"/dev/hidraw0","serialNumber":"","manufacturer":"Ultimate Gadget Laboratories","product":"UHK 60 v1","release":258,"interface":0,"usagePage":65280,"usage":1}
[UhkHidDevice] Device communication closing.
[UhkHidDevice] Device communication closed.
[DeviceOperation] Cannot ping the bootloader. Please remove the "Left keyboard half" module, and keep reconnecting it until you see this message.
[UhkHidDevice] Available devices unchanged
[UhkHidDevice] Used device: {"vendorId":"0x1D50","productId":"0x6122","path":"/dev/hidraw0","serialNumber":"","manufacturer":"Ultimate Gadget Laboratories","product":"UHK 60 v1","release":258,"interface":0,"usagePage":65280,"usage":1}
[UhkHidDevice] Device communication closing.
[UhkHidDevice] Device communication closed.
[DeviceOperation] Cannot ping the bootloader. Please remove the "Left keyboard half" module, and keep reconnecting it until you see this message.
[UhkHidDevice] Available devices unchanged
[UhkHidDevice] Used device: {"vendorId":"0x1D50","productId":"0x6122","path":"/dev/hidraw0","serialNumber":"","manufacturer":"Ultimate Gadget Laboratories","product":"UHK 60 v1","release":258,"interface":0,"usagePage":65280,"usage":1}
[UhkHidDevice] Device communication closing.
[UhkHidDevice] Device communication closed.
[DeviceOperation] Cannot ping the bootloader. Please remove the "Left keyboard half" module, and keep reconnecting it until you see this message.
[UhkHidDevice] Available devices unchanged
[UhkHidDevice] Used device: {"vendorId":"0x1D50","productId":"0x6122","path":"/dev/hidraw0","serialNumber":"","manufacturer":"Ultimate Gadget Laboratories","product":"UHK 60 v1","release":258,"interface":0,"usagePage":65280,"usage":1}
[UhkHidDevice] Device communication closing.
[UhkHidDevice] Device communication closed.
[DeviceOperation] Cannot ping the bootloader. Please remove the "Left keyboard half" module, and keep reconnecting it until you see this message.
[UhkHidDevice] Available devices unchanged
[UhkHidDevice] Used device: {"vendorId":"0x1D50","productId":"0x6122","path":"/dev/hidraw0","serialNumber":"","manufacturer":"Ultimate Gadget Laboratories","product":"UHK 60 v1","release":258,"interface":0,"usagePage":65280,"usage":1}
[UhkHidDevice] Device communication closing.
[UhkHidDevice] Device communication closed.
[DeviceOperation] Cannot ping the bootloader. Please remove the "Left keyboard half" module, and keep reconnecting it until you see this message.
[UhkHidDevice] Available devices unchanged
[UhkHidDevice] Used device: {"vendorId":"0x1D50","productId":"0x6122","path":"/dev/hidraw0","serialNumber":"","manufacturer":"Ultimate Gadget Laboratories","product":"UHK 60 v1","release":258,"interface":0,"usagePage":65280,"usage":1}
[UhkHidDevice] Device communication closing.
[UhkHidDevice] Device communication closed.
[DeviceOperation] Cannot ping the bootloader. Please remove the "Left keyboard half" module, and keep reconnecting it until you see this message.
[UhkHidDevice] Available devices unchanged
[UhkHidDevice] Used device: {"vendorId":"0x1D50","productId":"0x6122","path":"/dev/hidraw0","serialNumber":"","manufacturer":"Ultimate Gadget Laboratories","product":"UHK 60 v1","release":258,"interface":0,"usagePage":65280,"usage":1}
[UhkHidDevice] Device communication closing.
[UhkHidDevice] Device communication closed.
[DeviceOperation] Cannot ping the bootloader. Please remove the "Left keyboard half" module, and keep reconnecting it until you see this message.
[UhkHidDevice] Available devices unchanged
[UhkHidDevice] Used device: {"vendorId":"0x1D50","productId":"0x6122","path":"/dev/hidraw0","serialNumber":"","manufacturer":"Ultimate Gadget Laboratories","product":"UHK 60 v1","release":258,"interface":0,"usagePage":65280,"usage":1}
[UhkHidDevice] Device communication closing.
[UhkHidDevice] Device communication closed.
[DeviceOperation] Cannot ping the bootloader. Please remove the "Left keyboard half" module, and keep reconnecting it until you see this message.
[UhkHidDevice] Available devices unchanged
[UhkHidDevice] Used device: {"vendorId":"0x1D50","productId":"0x6122","path":"/dev/hidraw0","serialNumber":"","manufacturer":"Ultimate Gadget Laboratories","product":"UHK 60 v1","release":258,"interface":0,"usagePage":65280,"usage":1}
[UhkHidDevice] Device communication closing.
[UhkHidDevice] Device communication closed.
[DeviceOperation] Cannot ping the bootloader. Please remove the "Left keyboard half" module, and keep reconnecting it until you see this message.
[UhkHidDevice] Available devices unchanged
[UhkHidDevice] Used device: {"vendorId":"0x1D50","productId":"0x6122","path":"/dev/hidraw0","serialNumber":"","manufacturer":"Ultimate Gadget Laboratories","product":"UHK 60 v1","release":258,"interface":0,"usagePage":65280,"usage":1}
[UhkHidDevice] Device communication closing.
[UhkHidDevice] Device communication closed.
[DeviceOperation] Cannot ping the bootloader. Please remove the "Left keyboard half" module, and keep reconnecting it until you see this message.
[UhkHidDevice] Available devices unchanged
[UhkHidDevice] Used device: {"vendorId":"0x1D50","productId":"0x6122","path":"/dev/hidraw0","serialNumber":"","manufacturer":"Ultimate Gadget Laboratories","product":"UHK 60 v1","release":258,"interface":0,"usagePage":65280,"usage":1}
[UhkHidDevice] Device communication closing.
[UhkHidDevice] Device communication closed.
[DeviceOperation] Cannot ping the bootloader. Please remove the "Left keyboard half" module, and keep reconnecting it until you see this message.
[UhkOperations] Couldn't connect to the "Left keyboard half".
[DeviceService] updateFirmware error {"message":"[UhkOperations] Couldn't connect to the \"Left keyboard half\".","stack":"Error: [UhkOperations] Couldn't connect to the \"Left keyboard half\".\n    at t.UhkOperations.<anonymous> (/tmp/.mount_UHK.AgDHkQKb/resources/app.asar/electron-main.js:2:1246994)\n    at Generator.next (<anonymous>)\n    at s (/tmp/.mount_UHK.AgDHkQKb/resources/app.asar/electron-main.js:2:1303055)"}
[UhkHidDevice] Available devices unchanged
[UhkHidDevice] Used device: {"vendorId":"0x1D50","productId":"0x6122","path":"/dev/hidraw0","serialNumber":"","manufacturer":"Ultimate Gadget Laboratories","product":"UHK 60 v1","release":258,"interface":0,"usagePage":65280,"usage":1}
[DeviceService] start poll UHK Device
[UhkOperations] Couldn't connect to the "Left keyboard half".

**Comment by mondalaci** [comment]
Please retry with multiple computers, without modules, without KVM switch / USB hub, docking station, and give a try to every USB port.

**Comment by svenknoke** [comment]
Hi Lazlo,

I did as you asked, I even changed the USB Cable of the UHK. Same result. Left Half still dead and not connecting.

What else can I try. Missing half of the KeyBoard is no option and I really love my UHK.


Regards Sven

**Comment by mondalaci** [comment]
Please shoot an email to support@ultimatehackingkeyboard.com and we'll recall your UHK for repair.

**Comment by benjaminmodayil** [comment]
Yep. I'm having the same issue for my UHK V1 and it generally occurred the same time as @svenknoke's error occurred. I'm seeing the same logs when trying to update as well.

Problem is that I had **two** UHK V1's and when the first one's left module went dead, I thought it was just the keyboard, but when plugging in the second one, **the left module for the second went dead as well**. Tried plugging into an Intel Mac, M1 Max Mac, and a Windows computer, but still the same issue.

The issue came about when I opened the UHK software and was trying to update the keyboard to the latest firmware because I noticed some weird "clicking" behavior and wasn't sure if it was the keyboard or related to another app on my computer. I can't remember if it went dead upon opening the software or when I tried updating the firmware.

I'll also send an email to that support link...
